packages/editor/src/components/post-trash/check.js

```diff
@@ -1,20 +1,15 @@
 /**
  * WordPress dependencies
  */
-import { withSelect } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
 
-function PostTrashCheck( { isNew, postId, children } ) {
+export default function PostTrashCheck( { children } ) {
+	const { isEditedPostNew, getCurrentPostId } = useSelect( 'core/editor' );
```

Comment:
Convention for hooks is kinda broken here, so I think I'll have to make sure that we return the custom enhanced selector hooks with the `use` prefix.  So this would become something like:

```js
const { useIsEditedPostNew, useGetCurrentPostId } = useSelect( 'core/editor' );
```

Necessary because otherwise it won't be obvious (without knowing the internals ) that these are in fact react hooks and thus the [rules of Hooks](https://reactjs.org/docs/hooks-rules.html) apply to them.